packages/pages/src/components/map/map.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { createContext, useContext, useEffect, useRef, useState } from "react";
+import classnames from "classnames";
+import { GoogleMaps, Map as MapType, MapOptions } from "@yext/components-tsx-maps";
+import { Coordinate, GeoBounds } from "@yext/components-tsx-geo";
+import { MapProps, MapContextType } from "./types";
+import "./map.css";
+
+export const MapContext = createContext<MapContextType | null>(null);
+
+export function useMapContext() {
+  const ctx = useContext(MapContext);
+
+  if (!ctx || ctx.map === undefined) {
+    throw new Error('Attempted to call useMapContext() outside of <Map>.');
+  }
+
+  return ctx.map;
+}
+
+export const Map = ({
```

Comment:
The internals of the map api wrapper is a singleton class so it will only load the specific provider code once. Then it just creates a new instance of the map in each wrapper. I created a storybook where two maps are loaded.